lib/routemap.c

```diff
@@ -1664,7 +1664,24 @@ static struct route_map_index *route_map_get_index(struct route_map *map,
 	struct route_map_index *index = NULL, *best_index = NULL;
 	struct route_map_index *head_index = NULL;
 	struct route_table *table = NULL;
-	unsigned char family = prefix->family;
+	struct prefix conv;
+	unsigned char family;
+
+	/*
+	 * Handling for matching evpn_routes in the prefix table.
+	 *
+	 * We convert type2/5 prefix to ipv4/6 prefix to do longest
+	 * prefix matching on.
+	 */
+	if (prefix->family == AF_EVPN) {
```

Comment:
I think it's more readable as is because you will still need to do that below conditional after converting it.

So with your suggestion (as i understand it) it would end up like this:

```
if (family == evpn) {
     evpn_prefix_convert()
     ....
     if (con_pfx->family == AF_INET) {
     ...
     } else {
     ...
} else if (family == AF_INET) {
     ...
} else {
     ...
}
```

which looks worse imo.

Reply:
No. This is not what I meant, but I actually misunderstood this a bit. So this code is good.